src/bindings/text-buffer-wrapper.cc

```diff
@@ -708,7 +710,16 @@ class SaveWorker : public Nan::AsyncWorker {
     AsyncWorker(completion_callback),
     snapshot{snapshot},
     file_name{file_name},
-    encoding_name(encoding_name) {}
+    encoding_name(encoding_name),
+    file{open_file(file_name, "wb+")} {}
+
+  SaveWorker(Nan::Callback *completion_callback, TextBuffer::Snapshot *snapshot,
+             int file_descriptor, string &&encoding_name) :
+    AsyncWorker(completion_callback),
+    snapshot{snapshot},
+    file_name{"<file descriptor " + std::to_string(file_descriptor) + ">"},
+    encoding_name(encoding_name),
+    file{fdopen(file_descriptor, "wb+")} {}
```

Comment:
Really nice reuse here. I would think that we would want to avoid calling `fclose` on the file in this case though, so that we don't close the file descriptor. Alternatively, maybe we could use `dup` to clone the file descriptor passed to `SaveWorker`?